components/ws-manager/pkg/manager/monitor.go

```diff
@@ -440,20 +440,8 @@ func actOnPodEvent(ctx context.Context, m actingManager, status *api.WorkspaceSt
 			return nil
 		}
 
-		var terminated bool
-		for _, c := range wso.Pod.Status.ContainerStatuses {
-			if c.Name == "workspace" {
-				// Note: sometimes container don't enter `terminated`, but `waiting`. The processes are stopped nonetheless,
-				//       and we should be running the backup. The only thing that keeps the pod alive, is our finalizer.
-				terminated = c.State.Running == nil
-				break
-			}
-		}
-
-		_, gone := wso.Pod.Annotations[wsk8s.ContainerIsGoneAnnotation]
 		_, alreadyFinalized := wso.Pod.Annotations[startedDisposalAnnotation]
-
-		if (terminated || gone) && !alreadyFinalized {
+		if !alreadyFinalized {
```

Comment:
I am concerned about removing `terminated` check here. 
We want to make sure that we start backup ONLY when all processes have shutdown and container is stopped, otherwise we are risking trying to make a backup while something is terminating, and thus potentially making unstable backup.

But I guess in the case when node is not ready, k8s has no way to check the status of the container to ensure that it was stopped, hence it is stuck in fake running state.
What if we check the status of node here instead? And if it is not ready, then ignore terminated check and proceed making backup?

Reply:
>You mean a case where the node is NotReady and the Pod is still in Running state (haven't changed to Terminating state), and the user manually stops the workspace? I could test this scenario as well.

No, I am referring to a normal case: workspace is stopping. We wait until container is stopped, and only then start backup. 
If you remove `terminated` check here, as far as I understand it would mean we will attempt to run a backup right after pod transitions into terminating state, even though container is still running and is terminating\shutting down.
Or am I wrong here?

Reply:
Yes, you're right. The Pod is terminating; it executes the preStop hook (if it existed), and then sends a TERM signal to the container.

If during the workspace Pod graceful shutdown period, it generates more content, and either ws-daemon _or_ PVC snapshot finished the content backup, then the unstable backup happens.